paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentOptionFactory.kt

```diff
@@ -48,6 +50,12 @@ internal class PaymentOptionFactory @Inject constructor(
             loadIcon(darkThemeIconUrl)
         } else if (lightThemeIconUrl != null) {
             loadIcon(lightThemeIconUrl)
+        } else if (paymentOption.drawableResourceId == MISSING_DRAWABLE_RESOURCE) {
+            errorReporter.report(
+                ErrorReporter.UnexpectedErrorEvent.PAYMENT_OPTION_MISSING_ICON_URL_AND_RES,
+                additionalNonPiiParams = mapOf("payment_option" to paymentOption.label)
+            )
+            throw IllegalStateException("Missing icon resource and icon URLs - can't load")
```

Comment:
This feels sad, but the drawableResourceId in this class is exposed as part of our public API (but is deprecated), so I can't change the type of it. This case shouldn't ever happen though -- we always have either a resource OR an icon url.